### PR TITLE
🐛  Embed data files into the go lib files

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Embeds the data files (config.yaml and projects.csv) directly into the libraries which need and use it. 

* **What is the current behavior?** (You can also link to an open issue here)
The PubSub data files were specified with relative path, implying that they would be read relative to the dir where `go run` was called. This is very brittle and can break easily.

* **What is the new behavior (if this is a feature change)?**
The data files are embedded directly into the libraries that use them. There is also the option exposed to users to inject their own files if need be.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: cron/config/config_test.go [hunk 1]
```diff
@@ -29,6 +31,23 @@ const (
 	prodShardSize    int    = 250
 )
 
+func getByteValueFromFile(filename string) ([]byte, error) {
```
Comment (oliverchang) [inline cron/config/config_test.go:34]: Why not `ioutil.ReadFile()` ?

**Review by oliverchang — APPROVED** [review]

File: cron/config/config_test.go (hunk 1 above)
Comment (azeemshaikh38) [reply cron/config/config_test.go:33]: Nice, thanks. Still kept this fn though to check for the case where filename is empty.

**Comment by naveensrinivasan** [comment]
The idea of tests files is to make the data swappable at runtime. 

What was the reason for embedding?

**Comment by azeemshaikh38** [comment]
> The idea of tests files is to make the data swappable at runtime.
> 
> What was the reason for embedding?

Reason for embedding is so that the data filepaths are not dependent on where the `go run` command is executed from.

I have refactored such that data is still swappable during tests to as much extent as possible.

**Pull request merged**